File: Testing/AngularTest/TestApi3/src/web/rest/misc.controller.ts
import { Controller, Logger, Get, Req, Request, Query, Param, Post, Headers, Body } from '@nestjs/common';
import { PersonDTO } from './../../services/dto/People/person.dto';
import { PersonUpdateDTO } from './../../services/dto/People/person-update.dto';
import { DateDTO } from './../../services/dto/Misc/date.dto';
import { MiscService } from './../../services/misc.service';
import { JsonResponse } from './json-response';
import { ApiTags, ApiOkResponse, ApiBadRequestResponse, ApiNotFoundResponse, ApiCreatedResponse } from '@nestjs/swagger';

@ApiTags('Misc')
@Controller('api/misc')
export class MiscController {
  logger = new Logger('MiscController');

  constructor(private readonly miscService: MiscService) {}

  @Get('getwithqueryparam')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: PersonDTO,
  })
  @ApiBadRequestResponse({ description: 'Bad request.' })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithQueryParam(@Req() req: Request, @Query('idParam') idParam: string): Promise<PersonDTO> {
    const result = await this.miscService.getWithQueryParam(idParam);
    return result;
  }

  @Get('getwithrouteparam/:routeId')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: PersonDTO,
  })
  @ApiBadRequestResponse({ description: 'Bad request.' })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithRouteParam(@Req() req: Request, @Param('routeId') routeId: string): Promise<PersonDTO> {
    const result = await this.miscService.getWithRouteParam(routeId);
    return result;
  }

  @Post('postwithheaderparam')
  @ApiCreatedResponse({
    description: 'The record has been successfully created.',
  })
  @ApiBadRequestResponse({ description: 'Bad request.' })
  async postWithHeaderParam(@Req() req: Request, @Headers('param') param: string): Promise<void> {
    return await this.miscService.postWithHeaderParam(param);
  }

  @Post('postwithbodyparam')
  @ApiCreatedResponse({
    description: 'The record has been successfully created.',
  })
  @ApiBadRequestResponse({ description: 'Bad request.' })
  async postWithBodyParam(@Req() req: Request, @Body() param: PersonUpdateDTO): Promise<void> {
    return await this.miscService.postWithBodyParam(param);
  }

  @Get('getwithprimitiveresultint')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: 'number',
  })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithPrimitiveResultInt(@Req() req: Request): Promise<number> {
    const result = await this.miscService.getWithPrimitiveResultInt();
    return result;
  }

  @Get('getwithprimitiveresultwrapint')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: 'number',
  })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithPrimitiveResultWrapInt(@Req() req: Request): Promise<JsonResponse<number>> {
    const result = await this.miscService.getWithPrimitiveResultWrapInt();
    return new JsonResponse<number>(result);
  }

  @Get('getwithprimitiveresultbool')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: 'boolean',
  })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithPrimitiveResultBool(@Req() req: Request): Promise<boolean> {
    const result = await this.miscService.getWithPrimitiveResultBool();
    return result;
  }

  @Get('getwithprimitiveresultwrapbool')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: 'boolean',
  })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithPrimitiveResultWrapBool(@Req() req: Request): Promise<JsonResponse<boolean>> {
    const result = await this.miscService.getWithPrimitiveResultWrapBool();
    return new JsonResponse<boolean>(result);
  }

  @Get('getwithprimitiveresultstr')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: 'string',
  })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithPrimitiveResultStr(@Req() req: Request): Promise<string> {
    const result = await this.miscService.getWithPrimitiveResultStr();
    return result;
  }

  @Get('getwithprimitiveresultwrapstr')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: 'string',
  })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithPrimitiveResultWrapStr(@Req() req: Request): Promise<JsonResponse<string>> {
    const result = await this.miscService.getWithPrimitiveResultWrapStr();
    return new JsonResponse<string>(result);
  }

  @Get('getwithprimitiveresultdate')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: 'Date',
  })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithPrimitiveResultDate(@Req() req: Request): Promise<Date> {
    const result = await this.miscService.getWithPrimitiveResultDate();
    return result;
  }

  @Get('getwithprimitiveresultwrapdate')
  @ApiOkResponse({
    description: 'Result retrieved successfully.',
    type: 'Date',
  })
  @ApiNotFoundResponse({ description: 'Response not found.' })
  async getWithPrimitiveResultWrapDate(@Req() req: Request): Promise<JsonResponse<Date>> {
    const result = await this.miscService.getWithPrimitiveResultWrapDate();
    return new JsonResponse<Date>(result);
  }

  @Post('postdateparam')
  @ApiCreatedResponse({
    description: 'The record has been successfully created.',
  })
  @ApiBadRequestResponse({ description: 'Bad request.' })
  async postDateParam(@Req() req: Request, @Query('date') date: Date, @Query('datetime') datetime: Date): Promise<void> {
    return await this.miscService.postDateParam(date, datetime);
  }

  @Post('postdateparamdto')
  @ApiCreatedResponse({
    description: 'The record has been successfully created.',
  })
  @ApiBadRequestResponse({ description: 'Bad request.' })
  async postDateParamDto(@Req() req: Request, @Body() dto: DateDTO): Promise<void> {
    return await this.miscService.postDateParamDto(dto);
  }
}

File: Testing/AngularTest/TestApi/TestApi.Api/Controllers/MiscController.cs
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;
using System.Transactions;
using Intent.RoslynWeaver.Attributes;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using TestApi.Api.Controllers.ResponseTypes;
using TestApi.Application.Interfaces;
using TestApi.Application.Misc;
using TestApi.Application.People;
using TestApi.Domain.Common.Interfaces;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: DefaultIntentManaged(Mode.Fully, Targets = Targets.Usings)]
[assembly: IntentTemplate("Intent.AspNetCore.Controllers.Controller", Version = "1.0")]

namespace TestApi.Api.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class MiscController : ControllerBase
    {
        private readonly IMiscService _appService;

        private readonly IUnitOfWork _unitOfWork;

        public MiscController(IMiscService appService,
            IUnitOfWork unitOfWork)
        {
            _appService = appService ?? throw new ArgumentNullException(nameof(appService));

            _unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(unitOfWork));
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified PersonDTO.</response>
        /// <response code="400">One or more validation errors have occurred.</response>
        /// <response code="404">Can't find an PersonDTO with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(PersonDTO), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status400BadRequest)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<PersonDTO>> GetWithQueryParam([FromQuery] Guid idParam, CancellationToken cancellationToken)
        {
            var result = default(PersonDTO);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithQueryParam(idParam);

            }
            return Ok(result);
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified PersonDTO.</response>
        /// <response code="400">One or more validation errors have occurred.</response>
        /// <response code="404">Can't find an PersonDTO with the parameters provided.</response>
        [HttpGet("[action]/{routeId}")]
        [ProducesResponseType(typeof(PersonDTO), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status400BadRequest)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<PersonDTO>> GetWithRouteParam([FromRoute] Guid routeId, CancellationToken cancellationToken)
        {
            var result = default(PersonDTO);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithRouteParam(routeId);

            }
            return Ok(result);
        }

        /// <summary>
        /// </summary>
        /// <response code="201">Successfully created.</response>
        /// <response code="400">One or more validation errors have occurred.</response>
        [HttpPost("[action]")]
        [ProducesResponseType(StatusCodes.Status201Created)]
        [ProducesResponseType(StatusCodes.Status400BadRequest)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult> PostWithHeaderParam([FromHeader(Name = "MY_HEADER")] string param, CancellationToken cancellationToken)
        {
            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                await _appService.PostWithHeaderParam(param);

                await _unitOfWork.SaveChangesAsync(cancellationToken);
                transaction.Complete();
            }
            return Created(string.Empty, null);
        }

        /// <summary>
        /// </summary>
        /// <response code="201">Successfully created.</response>
        /// <response code="400">One or more validation errors have occurred.</response>
        [HttpPost("[action]")]
        [ProducesResponseType(StatusCodes.Status201Created)]
        [ProducesResponseType(StatusCodes.Status400BadRequest)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult> PostWithBodyParam([FromBody] PersonUpdateDTO param, CancellationToken cancellationToken)
        {
            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                await _appService.PostWithBodyParam(param);

                await _unitOfWork.SaveChangesAsync(cancellationToken);
                transaction.Complete();
            }
            return Created(string.Empty, null);
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified int.</response>
        /// <response code="404">Can't find an int with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(int), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<int>> GetWithPrimitiveResultInt(CancellationToken cancellationToken)
        {
            var result = default(int);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithPrimitiveResultInt();

            }
            return Ok(result);
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified int.</response>
        /// <response code="404">Can't find an int with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(JsonResponse<int>), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<int>> GetWithPrimitiveResultWrapInt(CancellationToken cancellationToken)
        {
            var result = default(int);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithPrimitiveResultWrapInt();

            }
            return Ok(new JsonResponse<int>(result));
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified bool.</response>
        /// <response code="404">Can't find an bool with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<bool>> GetWithPrimitiveResultBool(CancellationToken cancellationToken)
        {
            var result = default(bool);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithPrimitiveResultBool();

            }
            return Ok(result);
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified bool.</response>
        /// <response code="404">Can't find an bool with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(JsonResponse<bool>), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<bool>> GetWithPrimitiveResultWrapBool(CancellationToken cancellationToken)
        {
            var result = default(bool);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithPrimitiveResultWrapBool();

            }
            return Ok(new JsonResponse<bool>(result));
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified string.</response>
        /// <response code="404">Can't find an string with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<string>> GetWithPrimitiveResultStr(CancellationToken cancellationToken)
        {
            var result = default(string);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithPrimitiveResultStr();

            }
            return Ok(result);
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified string.</response>
        /// <response code="404">Can't find an string with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(JsonResponse<string>), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<string>> GetWithPrimitiveResultWrapStr(CancellationToken cancellationToken)
        {
            var result = default(string);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithPrimitiveResultWrapStr();

            }
            return Ok(new JsonResponse<string>(result));
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified DateTime.</response>
        /// <response code="404">Can't find an DateTime with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(DateTime), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<DateTime>> GetWithPrimitiveResultDate(CancellationToken cancellationToken)
        {
            var result = default(DateTime);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithPrimitiveResultDate();

            }
            return Ok(result);
        }

        /// <summary>
        /// </summary>
        /// <response code="200">Returns the specified DateTime.</response>
        /// <response code="404">Can't find an DateTime with the parameters provided.</response>
        [HttpGet("[action]")]
        [ProducesResponseType(typeof(JsonResponse<DateTime>), StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<DateTime>> GetWithPrimitiveResultWrapDate(CancellationToken cancellationToken)
        {
            var result = default(DateTime);

            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                result = await _appService.GetWithPrimitiveResultWrapDate();

            }
            return Ok(new JsonResponse<DateTime>(result));
        }

        /// <summary>
        /// </summary>
        /// <response code="201">Successfully created.</response>
        /// <response code="400">One or more validation errors have occurred.</response>
        [HttpPost("[action]")]
        [ProducesResponseType(StatusCodes.Status201Created)]
        [ProducesResponseType(StatusCodes.Status400BadRequest)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult> PostDateParam([FromQuery] DateTime date, [FromQuery] DateTime datetime, CancellationToken cancellationToken)
        {
            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                await _appService.PostDateParam(date, datetime);

                await _unitOfWork.SaveChangesAsync(cancellationToken);
                transaction.Complete();
            }
            return Created(string.Empty, null);
        }

        /// <summary>
        /// </summary>
        /// <response code="201">Successfully created.</response>
        /// <response code="400">One or more validation errors have occurred.</response>
        [HttpPost("[action]")]
        [ProducesResponseType(StatusCodes.Status201Created)]
        [ProducesResponseType(StatusCodes.Status400BadRequest)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult> PostDateParamDto([FromBody] DateDTO dto, CancellationToken cancellationToken)
        {
            using (var transaction = new TransactionScope(TransactionScopeOption.Required,
                new TransactionOptions() { IsolationLevel = IsolationLevel.ReadCommitted }, TransactionScopeAsyncFlowOption.Enabled))
            {

                await _appService.PostDateParamDto(dto);

                await _unitOfWork.SaveChangesAsync(cancellationToken);
                transaction.Complete();
            }
            return Created(string.Empty, null);
        }


    }
}